src/main/java/edu/njit/jerse/ashe/utils/JavaCodeParser.java

```diff
@@ -98,34 +101,42 @@ public record MethodSignature(
-     * @return an Optional containing {@link MethodSignature} if found, else empty.
+     * @param method the method string from which to extract the signature
+     * @return {@link MethodSignature} representing the extracted method signature
      */
-    public static Optional<MethodSignature> extractMethodSignature(String method) {
+    public static MethodSignature extractMethodSignature(String method) {
         try {
             CompilationUnit cu = StaticJavaParser.parse(method);
-
             Optional<MethodDeclaration> methodDeclarationOpt = cu.findFirst(MethodDeclaration.class);
-            if (methodDeclarationOpt.isPresent()) {
-                MethodDeclaration methodDeclaration = methodDeclarationOpt.get();
 
-                String returnType = methodDeclaration.getType().asString();
-                String name = methodDeclaration.getName().asString();
-                String params = extractParameters(methodDeclaration);
-                List<Modifier.Keyword> modifiers = extractModifiers(methodDeclaration);
-                ModifierPresent presence = determineModifierPresence(modifiers);
+            // Explicitly check if the method declaration is present
+            if (methodDeclarationOpt.isEmpty()) {
+                errorMessage = "Invalid method string";
+                LOGGER.error(errorMessage);
+                throw new NoSuchElementException(errorMessage);
+            }
 
-                LOGGER.debug("Extracted method signature: Modifiers={} ReturnType={} Name={} Parameters={}",
-                        modifiers.isEmpty() ? "None" : modifiers.stream().map(Enum::name).collect(Collectors.joining(", ")), returnType, name, params);
+            MethodDeclaration methodDeclaration = methodDeclarationOpt.get();
+            String returnType = methodDeclaration.getType().asString();
+            String name = methodDeclaration.getName().asString();
+            String params = extractParameters(methodDeclaration);
+            List<Modifier.Keyword> modifiers = extractModifiers(methodDeclaration);
+            ModifierPresent presence = determineModifierPresence(modifiers);
 
-                return Optional.of(new MethodSignature(presence, modifiers, returnType, name, params));
-            }
+            LOGGER.info("Extracted method signature: Modifiers={} ReturnType={} Name={} Parameters={}",
+                    modifiers.isEmpty() ? "None" : modifiers.stream().map(Enum::name).collect(Collectors.joining(", ")),
+                    returnType, name, params);
+
+            return new MethodSignature(presence, modifiers, returnType, name, params);
         } catch (ParseProblemException ex) {
-            LOGGER.error("Failed to parse method due to syntax error(s): {}", ex.getProblems());
+            errorMessage = "Failed to parse method due to syntax error(s): " + ex.getProblems();
+            LOGGER.error(errorMessage);
+            throw new RuntimeException(errorMessage);
```

Comment:
Rather than catching `ex` and then throwing a different exception, I'd prefer we just rethrow `ex` here (and in the other catch block below) after logging the error message.